docs/cookbook/import-categories.md

```diff
@@ -0,0 +1,198 @@
+# Import categories
+
+Category import is one of the specific imports.
+Due to the tree structure it's really easy to create ineffective script.
+In this cookbook we highlight some concepts and how to improve category import performance.
+
+## Task
+
+> _As an eshop owner I want to import categories from IS, so I don't need to edit categories on the two places._
+
+Let's presume we already have working information system, and we are able to obtain data we need.
+For the future references we can imagine it as a method with following description
+
+```php
+public function getCategoryDataFromIs(): iterable {
+    // exact implementation is not important
+    // returns an iterable which each element looks like
+    //   [
+    //       'uuid' => '1bb2dd5d-92e0-4263-8ca3-d63aee840c62',
+    //       'parent_uuid' => '3af3c39a-d4b0-4a59-9e36-2e5d54e7ca21',
+    //       'name_en' => 'Electronics',
+    //       'name_cs' => 'Elektronika',
+    //       'description_1' => 'Description of Electronics',
+    //       'description_2' => 'Description of Electronics in Czech',
+    //   ];
+}
+```
+
+Method returns data sorted exactly as they should be on the eshop, meaning category with the same parent will be first, if it comes first from the method.  
+And to make matters worse, IS can only return all 10000 categories.
+
+## First import
```

Comment:
It was bit confusing for me, that I am following this guide and later I found, that there is better way. Maybe some info about it would be nice.

Reply:
🤔 that's right, it's not the best solution in the first paragraph, but I tried to put it in a "improve the code on the way", it's not "now delete everything and writ this", or?

Reply:
I have added the info in d20c184e8